### PR TITLE
Comment post

### PR DESCRIPTION
**Pull request by agrawalshanu708**
Added a feature: Comments
user can add comment to post
user can upvote/downvote a comment
user can edit own written comment
user can delete a comment

image

### CODE REVIEW

File: src/comps/comments/CommentPannel.jsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      downvotedComment({ postId: post._id, commentId: comment._id, token })
+    );
+  };
+
+  const upvoteUser = comment.votes.upvotedBy.find(
+    (el) => el._id === foundUser._id
+  );
+  const downvoteUser = comment.votes.downvotedBy.find(
+    (el) => el._id === foundUser._id
+  );
```
Comment (harshitpaliwal95) [inline src/comps/comments/CommentPannel.jsx:54]: can create single function

File: src/redux/slice/post/postServices.js [hunk 2]
```diff
@@ -14,68 +12,188 @@ export const getAllPost = createAsyncThunk(
...
+    } catch (error) {
+      return rejectWithValue(`Error from deletecomment: ${error.message}`);
+    }
+  }
+);
+
+export const editComment = createAsyncThunk(
+  "post/editComment",
+  async ({ postId, commentId, commentData, token }, { rejectWithValue }) => {
+    console.log(commentId,commentData)
```
Comment (vaibhav18matere) [inline src/redux/slice/post/postServices.js:187]: remove logs

File: src/redux/slice/post/postServices.js [hunk 3]
```diff
@@ -14,68 +12,188 @@ export const getAllPost = createAsyncThunk(
...
+    console.log(postId, token);
+    try {
+      const response = await axios.delete(
+        `/api/posts/${postId}`,
+        {},
+        { headers: { authorization: token } }
+      );
+      console.log(response);
+    } catch (error) {
+      console.log(error);
```
Comment (vaibhav18matere) [inline src/redux/slice/post/postServices.js:103]: remove this if not going to handle error on UI later

**Review by vaibhav18matere — APPROVED** [review]
looks pretty clean, just a small suggestion:

**Review by harshitpaliwal95 — COMMENTED** [review]
code looks good 
need to improve typo in comment comp

**Pull request merged**